### PR TITLE
Workaround for MS compiler bug

### PR DESCRIPTION
**Pull request by bellenot**
This is a workaround for the issue reported as [Another fatal error C1001: Internal compiler error](https://developercommunity.visualstudio.com/t/another-fatal-error-c1001-internal-compiler-error/1441527)

### CODE REVIEW

File: tree/dataframe/test/CMakeLists.txt [hunk 1]
```diff
@@ -43,6 +43,11 @@ if (imt)
    ROOT_ADD_GTEST(dataframe_concurrency dataframe_concurrency.cxx LIBRARIES ROOTDataFrame)
 endif()
 
+if (MSVC)
+  # TODO: remove this workaround for MS compiler bug #1441527 once fixed
+  string(REPLACE "-Od -Z7" "-O2" CMAKE_CXX_FLAGS_DEBUG "${CMAKE_CXX_FLAGS_DEBUG}")
+  string(REPLACE "-Z7" "" CMAKE_CXX_FLAGS_RELWITHDEBINFO "${CMAKE_CXX_FLAGS_RELWITHDEBINFO}")
+endif()
```
Comment (eguiraud) [inline tree/dataframe/test/CMakeLists.txt:50]: should the flags be restored to the original values afterwards? if not, should the flags be overridden at the project level?

Comment (bellenot) [reply tree/dataframe/test/CMakeLists.txt:50]: > should the flags be restored to the original values afterwards?

Well, yes, if it is possible. I have no experience. Let me try...

Comment (bellenot) [reply tree/dataframe/test/CMakeLists.txt:50]: You mean something like this:
```cmake
if (MSVC)
  # TODO: remove this workaround for MS compiler bug #1441527 once fixed
  string(REPLACE "-Od -Z7" "-O2" CMAKE_CXX_FLAGS_DEBUG "${CMAKE_CXX_FLAGS_DEBUG}")
  string(REPLACE "-Z7" "" CMAKE_CXX_FLAGS_RELWITHDEBINFO "${CMAKE_CXX_FLAGS_RELWITHDEBINFO}")
endif()
ROOT_ADD_GTEST(datasource_more datasource_more.cxx LIBRARIES ROOTDataFrame)
if (MSVC)
  # restore the compiler flags
  string(REPLACE "-O2" "-Od -Z7" CMAKE_CXX_FLAGS_DEBUG "${CMAKE_CXX_FLAGS_DEBUG}")
  set(CMAKE_CXX_FLAGS_RELWITHDEBINFO "${CMAKE_CXX_FLAGS_RELWITHDEBINFO} -Z7")
endif()
```

Comment (eguiraud) [reply tree/dataframe/test/CMakeLists.txt:50]: I guess so :smile: what do you think?

Comment (bellenot) [reply tree/dataframe/test/CMakeLists.txt:50]: Doesn't work. Still working on it

Comment (Axel-Naumann) [reply tree/dataframe/test/CMakeLists.txt:50]: You should be able to use [`set_source_files_properties`](https://cmake.org/cmake/help/latest/command/set_source_files_properties.html)?

Comment (bellenot) [reply tree/dataframe/test/CMakeLists.txt:50]: @Axel-Naumann it doesn't work

**Review by eguiraud — APPROVED** [review]
my cmake-fu is not strong enough, I would think this changes the flags for the rest of the project, but if that's not the case no problem on my side: whatever works for windows :sweat_smile: :sweat_smile:

**Review by Axel-Naumann — APPROVED** [review]

**Pull request merged**